File: 2023/java/src/test/java/org/frieder/aoc/day7/TestDay7A.java
package org.frieder.aoc.day7;

import org.frieder.aoc.day7.a.Solution7A;
import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.assertEquals;

public class TestDay7A {
    @Test
    public void testExample() throws IOException {
        assertEquals(6440, Solution7A.getResult("../resources/7/test_a.txt"), 0);
    }

    @Test
    public void testResult() throws IOException {
        assertEquals(251029473, Solution7A.getResult("../resources/7/input.txt"), 0);
    }
}

File: 2023/php/test/TestDay7A.php
<?php declare(strict_types=1);

$loader = require __DIR__ . '/../vendor/autoload.php';

use day7\a\Solution7A;
use PHPUnit\Framework\TestCase;
use utils\TestLogger;

final class TestDay7A extends TestCase
{

    public function testExample(): void
    {
        $this->assertSame(6440, Solution7A::getResult("../resources/7/test_a.txt", TestLogger::createTestLogger()));
    }

    public function testResult(): void
    {
        $this->assertSame(251029473, Solution7A::getResult("../resources/7/input.txt", TestLogger::createTestLogger()));
    }

}